nixos/modules/services/misc/persistent-evdev.nix

```diff
@@ -0,0 +1,73 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.persistent-evdev;
+  configFile = pkgs.writeText "persistent-evdev-config.json" ''
+    ${builtins.toJSON {
+      cache = "${cfg.cacheDir}";
+      devices = mapAttrs (virt: phys: "/dev/input/by-id/${phys}") cfg.devices;
+    }}
+  '';
+in
+{
+  options.services.persistent-evdev = {
+    enable = mkEnableOption "virtual input devices that persist even if the backing device is hotplugged";
+
+    cacheDir = mkOption {
+      default = "/var/cache/persistent-evdev";
+      type = types.path;
+      description = ''
+        Directory for caching physical device information so that virtual devices can persist even
+        if the underlying physical devices are disconnected and later reconnected.
+      '';
+    };
+
+    devices = mkOption {
+      default = {};
+      type = types.attrsOf types.str;
+      description = ''
+        A set of virtual proxy device labels with backing physical device ids.
+
+        Physical devices should already exist in /dev/input/by-id/.  Proxy devices will be created
+        with a "uinput-" prefix.
+
+        See https://github.com/aiberia/persistent-evdev#example-usage-with-libvirt for example
+        usage of virtual devices with libvirt.  Remember to add uinput-* devices to the qemu
+        cgroup_device_acl list (nixos: virtualisation.libvirtd.qemu.verbatimConfig).
```

Comment:
this is a docbook text, so links and paragraphs will have to use docbook formatting to show up in the manuals as you intended them to.